src/main/java/david/data/Storage.java

```diff
@@ -0,0 +1,143 @@
+package david.data;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import david.exceptions.DavidCacheException;
+import david.exceptions.DavidInvalidDateTimeException;
+import david.parser.DateParser;
+import david.task.DeadlineTask;
+import david.task.EventTask;
+import david.task.Task;
+import david.task.TaskList;
+import david.task.TodoTask;
```

Comment:
I like how there is a consistent ordering of import statements which makes it easier to browse the list and determine the dependencies when there are many imports, and also how you didn't use the wildcard for importing.